src/System.Private.ServiceModel/tests/Common/Infrastructure/src/BridgeClient.cs

```diff
@@ -83,16 +84,51 @@ public static string GetResourceAddress(string resourceName)
             string resourceAddress = null;
             lock (s_thisLock)
             {
-                if (s_BridgeStatus == BridgeState.Started && !_Resources.TryGetValue(resourceName, out resourceAddress))
+                Dictionary<string, string> resources = null;
+                if (s_BridgeStatus == BridgeState.Started)
                 {
-                    resourceAddress = MakeEndpointResourcePutRequest(resourceName);
-                    _Resources.Add(resourceName, resourceAddress);
+                    if (!_Resources.TryGetValue(resourceName, out resources))
+                    {
+                        resources = MakeEndpointResourcePutRequest(resourceName);
+                        resources.TryGetValue(EndpointResourceResponseUriKeyName, out resourceAddress);
+                        _Resources.Add(resourceName, resources);
```

Comment:
Nit: do this resource.TryGetValue just in one place, after the 'if' block.  Don't need the 'else' block.